### PR TITLE
HIVE-19880 : Repl Load to return recoverable vs non-recoverable error…

### PR DESCRIPTION
**Pull request by maheshk114**
…

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java [hunk 1]
```diff
@@ -228,14 +232,24 @@ private void analyzeReplDump(ASTNode ast) throws SemanticException {
...
       switch (childNode.getToken().getType()) {
         case TOK_DBNAME:
           dbNameOrPattern = PlanUtils.stripQuotes(childNode.getChild(0).getText());
+          for (String dbName : Utils.matchesDb(db, dbNameOrPattern)) {
+            Database database = db.getDatabase(dbName);
+            if (database != null) {
+              if (ReplChangeManager.isSourceOfReplication(database)) {
+                LOG.error("Cannot load database " + dbName +
+                        " as it is a source of replication");
+                throw new SemanticException(ErrorMsg.REPL_TARGET_IS_THE_SOURCE_OF_REPLICATION.getMsg());
```
Comment (sankarh) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java]: Why do we need to fail for this case? Isn't it enough to ensure, we don't overwrite this property when we load?

File: ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java [hunk 2]
```diff
@@ -145,7 +148,7 @@ private void doCopyRetry(FileSystem sourceFs, List<ReplChangeManager.FileInfo> s
     // If still files remains to be copied due to failure/checksum mismatch after several attempts, then throw error
     if (!pathList.isEmpty()) {
       LOG.error("File copy failed even after several attempts. Files list: " + pathList);
-      throw new IOException("File copy failed even after several attempts.");
+      throw new IOException(ErrorMsg.REPL_FILE_SYSTEM_RETRY.getMsg());
```
Comment (sankarh) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java]: This is File copy error not file system retry case.

File: standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java [hunk 3]
```diff
@@ -2712,7 +2715,7 @@ public NotificationEventResponse getNextNotification(long lastEventId, int maxEv
                   + "Try setting higher value for hive.metastore.event.db.listener.timetolive. "
                   + "Also, bootstrap the system again to get back the consistent replicated state.",
                   nextEventId, e.getEventId());
-          throw new IllegalStateException("Notification events are missing.");
+          throw new IllegalStateException(REPL_EVENTS_MISSING_IN_METASTORE);
```
Comment (sankarh) [inline standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java]: Even this is non-recoverable error. So, shouldn't this be FatalException?

File: standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java [hunk 4]
```diff
@@ -128,6 +128,9 @@
   private long retryDelaySeconds = 0;
   private final ClientCapabilities version;
 
+  //copied from ErrorMsg.java
+  private static final String REPL_EVENTS_MISSING_IN_METASTORE = "Notification events are missing in the meta store.";
```
Comment (sankarh) [inline standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java]: Why do we need this msg same as from ErrorMsg?

File: ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java (hunk 1 above)
Comment (maheshk114) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/ReplicationSemanticAnalyzer.java]: remove this error code

**Pull request closed without merging**